prepare_check_cleanup/migration.py

```diff
@@ -0,0 +1,269 @@
+    """
+    test_name = "check-branch"
+
+    github_repo = _create_repository_client(github_access_token=github_access_token)
+    migration_branch = _get_migration_branch(github_repo=github_repo)
+    if not migration_branch:
+        logging.error(
+            "%s check failed, migration branch %s not created", test_name, DEFAULT_BRANCH_NAME
+        )
+        return False
+
+    logging.info("%s check succeeded", test_name)
+    return True
+
+
+def check_pull_request(github_access_token: str) -> bool:
+    """Check that the migration pull request was created.
+
+    Args:
+        github_access_token: The secret required for interactions with GitHub.
+
+    Returns:
+        Whether the test succeeded.
+    """
+    test_name = "check-pull-request"
+
+    github_repo = _create_repository_client(github_access_token=github_access_token)
+    migration_pull_request = _get_migration_pull_request(github_repo=github_repo)
+    if not migration_pull_request:
+        logging.error(
+            "%s check failed, migration pull request %s not created",
+            test_name,
+            ACTIONS_PULL_REQUEST_TITLE,
+        )
+        return False
+
+    expected_files = {
+        f"{DOCUMENTATION_FOLDER_NAME}/{DOCUMENTATION_INDEX_FILENAME}",
+        f"{DOCUMENTATION_FOLDER_NAME}/page.md",
+    }
+    files_in_pull_request = {file.filename for file in migration_pull_request.get_files()}
+    if expected_files == files_in_pull_request:
+        logging.error(
+            "%s check failed, migration pull request does not contain expected files, "
+            "expected file: %s, files in pull request: %s",
+            test_name,
+            expected_files,
+            files_in_pull_request,
+        )
+        return False
+
+    logging.info("%s check succeeded", test_name)
+    return True
+
+
+def cleanup(
+    topics: dict[str, str], github_access_token: str, discourse_config: dict[str, str]
+) -> None:
+    """Clean up testing artifacts on GitHub and Discourse.
+
+    Args:
+        topics: The discourse topics created for the migration.
+        github_access_token: The secret required for interactions with GitHub.
+        discourse_config: Details required to communicate with discourse.
+    """
+    # Delete discourse topics
+    discourse = create_discourse(**discourse_config)
+    try:
+        for topic_url in topics.values():
+            discourse.delete_topic(url=topic_url)
+    except DiscourseError as exc:
+        logging.exception("cleanup failed for discourse, %s", exc)
+
+    github_repo = _create_repository_client(github_access_token=github_access_token)
+    # Delete the migration PR
+    try:
+        migration_pull_request = _get_migration_pull_request(github_repo=github_repo)
+        if migration_pull_request:
+            migration_pull_request.edit(state="closed")
+    except GithubException as exc:
+        logging.exception("cleanup failed for migration pull request, %s", exc)
```

Comment:
Hmm I'm wondering if deleting the PR branch doesn't automatically close the PR

Reply:
Yes it may, it also may complain that there is still an open PR. Either way, i think it is good to be explicit about cleaning up what was created, that way if GitHub changes something we don't suddenly have a whole bunch of PRs lying around